src/catalog/catalog_set.cpp

```diff
@@ -341,23 +340,24 @@ DuckCatalog &CatalogSet::GetCatalog() {
 
 void CatalogSet::CleanupEntry(CatalogEntry &catalog_entry) {
 	// destroy the backed up entry: it is no longer required
-	D_ASSERT(catalog_entry.parent);
-	if (catalog_entry.parent->type != CatalogType::UPDATED_ENTRY) {
+	auto parent_p = catalog_entry.Parent();
+	D_ASSERT(parent_p);
+	if (parent_p->type != CatalogType::UPDATED_ENTRY) {
 		lock_guard<mutex> write_lock(catalog.GetWriteLock());
 		lock_guard<mutex> lock(catalog_lock);
 		if (!catalog_entry.deleted) {
 			// delete the entry from the dependency manager, if it is not deleted yet
 			D_ASSERT(catalog_entry.ParentCatalog().IsDuckCatalog());
 			catalog_entry.ParentCatalog().Cast<DuckCatalog>().GetDependencyManager().EraseObject(catalog_entry);
 		}
-		auto parent = catalog_entry.parent;
-		parent->child = std::move(catalog_entry.child);
```

Comment:
I'm pretty sure this was a bug?
The `child` we're moving here never gets its `parent` set to the new parent